### DISCUSSION TITLE
Data migration due to removal of Application.Properties

### DISCUSSION THREAD

**Discussion by Jared-Aus** [dotnet/maui#6504] [General]
Given the issue [Proposal: Remove Application.Properties](https://github.com/dotnet/maui/issues/4864) has been closed, I started this discussion to state how brutal this change appears to be and hopefully get suggestions on what options might be available to migrate a user's existing preferences. I can think of a few, however none sound appealing. I get that there are plenty of advantages for fully killing off the legacy way of storing preferences, and migrating to Essentials is a trivial task if we don't care about user data.

In my experience, newer XF apps typically use Essentials for preferences, but older apps using Application.Properties didn't switch over because of the need for data migration, or simply because they are old and worked as intended.

@davidortinau - you suggested providing a good compiler error for this, but are you expecting the advice to extend to suggestions on how to preserve user data? I get that it's just an XML file, but without looking at the old implementation developers wouldn't know if there are any gotchas waiting to strike should they considering manually migrating data.

**Comment by greg84** [comment]
Agreed this needs sorting out. I’ve added a proposal for it: https://github.com/dotnet/maui/issues/14817

In the meantime I’ll likely release an update to the Xamarin app to migrate the properties dictionary to a custom solution.

**Comment by jfversluis** [comment]
Hey there! I have created a library that should help with this scenario. You can find it [here](https://github.com/jfversluis/Plugin.Maui.FormsMigration) (and on NuGet).

Additionally you can find guidance for this scenario in the Docs, [here](https://learn.microsoft.com/dotnet/maui/migration/app-properties).

I'd be curious to hear your findings!
